pkg/northbound/proto/admin.proto

```diff
@@ -19,6 +19,152 @@ syntax = "proto3";
 // Package admin defines the administrative and diagnostic gRPC interfaces.
 package proto;
 
+// AddDeviceRequest adds a device to the topology
+message AddDeviceRequest {
+
+}
+
+// AddDeviceResponse is sent in response to an AddDeviceRequest
+message AddDeviceResponse {
+
+}
+
+// GetDeviceRequest gets a device by ID
+message GetDeviceRequest {
+    string device_id = 1;
+}
+
+// GetDeviceResponse carries a device
+message GetDeviceResponse {
+    Device device = 1;
+}
+
+// GetDevicesRequest gets a list of devices in the topology
+message GetDevicesRequest {
+
+}
+
+// GetDevicesResponse carries a list of devices
+message GetDevicesResponse {
+    repeated Device devices = 1;
+}
+
+// RemoveDeviceRequest removes a device by ID
+message RemoveDeviceRequest {
+    string device_id = 1;
+}
+
+// RemoveDeviceResponse is sent in response to a RemoveDeviceRequest
+message RemoveDeviceResponse {
+
+}
+
+// StreamDevicesRequest requests a stream of device changes
+// By default, the request requests a stream of all device events that occur *after* the request is
+// received by the service. However, if `all` is `true`, the stream is prefixed by a device `ADD` event
+// for each existing device in the store.
+message StreamDevicesRequest {
+
```

Comment:
Could have a name filter maybe

Reply:
Are you suggesting within? If so, yes... among other criteria such as type, availability, etc.

Reply:
I think that’s fine. We should address filtering and device parameters separately. The `Device` doesn’t currently even include most of the parameters on which we’d want to filter, but it is consistent with what’s in onos-config currently. I think we should either keep it that way until we’ve integrated onos-config and onos-topo and then address the changes needed to `Device` and filtering, or if someone wants to propose a change to the `Device` fields we can merge this and then add that in another PR before the onos-config/onos-topo work.